File: output/GPT3.5-Turbo/xCodeEval_4000/C++/Python/74b6837a3d04a6f57b5df4915333408c.py
import random
import sys
from typing import List, Tuple

def can(n: int, s: str) -> bool:
    if n == 1:
        return True
    x = n // 2
    y = (n + 1) // 2
    if not can(x, s) or not can(y, s):
        return False
    a = 0
    b = 0
    for i in range(len(s)):
        if s[i] == '0':
            a += 1
        else:
            b += 1
        if a == x or b == y:
            break
    if a != x and b != y:
        return False
    return True

def gen(l: int, r: int, s: str, it: int, id: List[int], mas: List[int], e: List[List[int]]) -> None:
    if l == r:
        return
    m = (l + r - 1) // 2
    gen(l, m, s, it, id, mas, e)
    gen(m + 1, r, s, it, id, mas, e)
    a = l
    b = m + 1
    c = l
    while a <= m and b <= r:
        if s[it] == '0':
            mas[c] = id[a]
            a += 1
        else:
            mas[c] = id[b]
            b += 1
        c += 1
        it += 1
    while a <= m:
        mas[c] = id[a]
        a += 1
        c += 1
    while b <= r:
        mas[c] = id[b]
        b += 1
        c += 1
    for i in range(l, r + 1):
        id[i] = mas[i]
        if i > l:
            e[id[i - 1]].append(id[i])

def go(n: int, s: str) -> None:
    it = 0
    if not can(n, s) or it != len(s):
        return

    it = 0
    id = [i for i in range(n)]
    mas = [0] * n
    e = [[] for _ in range(n)]
    gen(0, n - 1, s, it, id, mas, e)
    assert it == len(s)
    p = [0] * n
    for i in range(n):
        p[id[i]] = i
    print(n)
    for i in p:
        print(i + 1, end=' ')
    print()
    sys.exit(0)

def solve() -> None:
    seg = [[0, 0] for _ in range(maxn)]
    for i in range(2, maxn):
        x = (i + 1) // 2
        y = i - x
        for t in range(2):
            seg[i][t] = seg[x][t] + seg[y][t]
        seg[i][0] += x
        seg[i][1] += y

    s = input().strip()
    counter = [0, 0]
    for c in s:
        counter[int(c)] += 1
    for n in range(maxn - 1, 0, -1):
        if seg[n][0] >= counter[0] and seg[n][1] >= counter[1]:
            go(n, s)

if __name__ == '__main__':
    solve()

File: output/GPT4o-Mini/xCodeEval_4000/C++/Java/74b6837a3d04a6f57b5df4915333408c.java
import java.util.*;
import java.io.*;

public class 74b6837a3d04a6f57b5df4915333408c_nc {
    static final int MAXN = 1003;
    static final int MD = 998244353;
    static String s;
    static int it;
    static List<Integer>[] e = new ArrayList[MAXN];
    static int[] id = new int[MAXN];
    static int[] mas = new int[MAXN];
    static int[] deg = new int[MAXN];
    static int[][] seg = new int[MAXN][2];

    static {
        for (int i = 0; i < MAXN; i++) {
            e[i] = new ArrayList<>();
        }
    }

    static boolean can(int n) {
        if (n == 1) return true;
        int x = n / 2, y = (n + 1) / 2;
        if (!can(x) || !can(y)) return false;
        int a = 0, b = 0;
        while (it < s.length()) {
            if (s.charAt(it) == '0') a++;
            else b++;
            it++;
            if (a == x || b == y) break;
        }
        return a == x || b == y;
    }

    static void gen(int l, int r) {
        if (l == r) return;
        int m = (l + r - 1) / 2;
        gen(l, m);
        gen(m + 1, r);
        int a = l, b = m + 1, c = l;
        while (a <= m && b <= r) {
            if (s.charAt(it) == '0') {
                mas[c++] = id[a++];
            } else {
                mas[c++] = id[b++];
            }
            it++;
        }
        while (a <= m) mas[c++] = id[a++];
        while (b <= r) mas[c++] = id[b++];
        for (int i = l; i <= r; i++) {
            id[i] = mas[i];
            if (i > l) {
                e[id[i - 1]].add(id[i]);
            }
        }
    }

    static void go(int n) {
        it = 0;
        if (!can(n) || it != s.length()) return;

        it = 0;
        for (int i = 0; i < n; i++) id[i] = i;
        gen(0, n - 1);
        assert it == s.length();
        int[] p = new int[n];
        for (int i = 0; i < n; i++) p[id[i]] = i;
        System.out.println(n);
        for (int i : p) System.out.print((i + 1) + " ");
        System.out.println();
        System.exit(0);
    }

    static void solve() {
        for (int i = 2; i < MAXN; i++) {
            int x = (i + 1) / 2, y = i - x;
            for (int t = 0; t < 2; t++) {
                seg[i][t] = seg[x][t] + seg[y][t];
            }
            seg[i][0] += x;
            seg[i][1] += y;
        }
        Scanner scanner = new Scanner(System.in);
        s = scanner.next();
        int[] counter = new int[2];
        for (char c : s.toCharArray()) counter[c - '0']++;
        for (int n = MAXN - 1; n > 0; n--) {
            if (seg[n][0] >= counter[0] && seg[n][1] >= counter[1]) {
                go(n);
            }
        }
    }

    public static void main(String[] args) {
        Scanner scanner = new Scanner(System.in);
        System.setIn(new BufferedInputStream(new FileInputStream("a.in")));
        System.setOut(new PrintStream(new FileOutputStream("a.out")));
        solve();
    }
}
